### ISSUE TITLE
Issue with improving extracted frames

### ISSUE THREAD

**Issue by beepboop174** [DeepLabCut/DeepLabCut#2552]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

Hello everyone!

I’ve created a maDLC model to track two separate mice, each with eight body parts. My plan is to use this model to quantify social interaction using SimBA. I’ve annotated around 150 frames, trained the model, and it works, kinda. I’m impressed by how it tracks the animals, but there are many instances where it loses sight of one of the two animals or loses track of the nose/tail base when they closely interact, which is crucial for the social interaction I’m trying to quantify. Train error is at around 3.1 and test error at 5.2, stalling at that level after 20.000 iterations. So, it definitely needs improvement. However, I’ve encountered some issues along the way.

Initially, I tried to extract outlier frames but faced problems with napari. In frames where the model didn’t identify a tail, for example, I couldn’t set a new marker for that body part. I could delete the markers it had set, but then I could only add markers for the body parts it had identified. If it couldn’t find the tail and I clicked to add that marker, nothing would appear, and it would move on to add the next body part it had identified. This makes it useless for improving frames where body parts are missing. How can I change this?

Afterward, I manually extracted 90 frames of close interaction from the training videos and labeled them, added them to my training set, and trained the model again. Here, I encountered my second issue: How do I create a second iteration of the model, improving it with added frames? I couldn’t find anything and simply hit “train,” but it seemed to start from scratch, and the new model is not better than my previous one, despite the addition of many frames of specific close interaction. So, I’m wondering, why didn’t it improve? Do I have to set it to shuffle 2? Also, any idea why, after evaluating the new model, the folder “LabeledImages_DLC_snapshot_newmodelnumber” is empty?

Lastly, I’m wondering whether downsampling videos may be a problem. The DLC walkthrough suggested doing it, and it does greatly expedite the analysis, but even I find it hard to distinguish body parts of low-resolution black and white images of closely interacting black mice. It makes sense that the model has issues as well. Would using higher quality video files help here?

Sorry for the ramble; this is a super cool tool, and creating a model was surprisingly doable for someone with no experience. But now, improving it to be actually usable is proving difficult.

Thanks for any help!

### Operating System

Windows 11

### DeepLabCut version

dlc version newest

### DeepLabCut mode

multi animal

### Device type

Intel core I9

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by n-poulsen** [comment]
Hi @beepboop174!

> Initially, I tried to extract outlier frames but faced problems with napari. In frames where the model didn’t identify a tail, for example, I couldn’t set a new marker for that body part. I could delete the markers it had set, but then I could only add markers for the body parts it had identified.

We've noticed this as well and just fixed it in [napari-deeplabcut #121](https://github.com/DeepLabCut/napari-deeplabcut/pull/121). This was just released in the newest version of napari-deeplabcut, which you can install by running `pip install "napari-deeplabcut==0.2.1.6"` or `pip install --upgrade napari-deeplabcut` in your conda environment.

> How do I create a second iteration of the model, improving it with added frames?

Once you've corrected the outlier frames and added missed detections, you can call [`deeplabcut.merge_datasets`](https://deeplabcut.github.io/DeepLabCut/docs/standardDeepLabCut_UserGuide.html#n-refine-labels-augmentation-of-the-training-dataset) (or in the GUI, click on the **_Merge data_** button in the **_Extract outlier frames_** tab). This increases your _project iteration_ by 1 (so if it was 0, it becomes 1). So in your `dlc-models`, `training-datasets` and `evaluation-results` folders, your models and data will be in the folder `iteration-1`. This _iteration_ is set in your project's `config.yaml` file, and everything you do from now on in the GUI will be for this iteration.

It's a good idea to go to the _Label Frames_ tab and click `Check Labels`, and verify that your labeled data looks correct (this is documented [here](https://deeplabcut.github.io/DeepLabCut/docs/maDLC_UserGuide.html#check-annotated-frames)).

You then need to recreate a training dataset (so in the DeepLabCut [user guide](https://deeplabcut.github.io/DeepLabCut/docs/maDLC_UserGuide.html#create-training-dataset), go back to the _Create Training Dataset(s)_ section). In the GUI, this just means going back to _Step 4. Create Training Dataset_. This will create a new shuffle for the new _iteration_. You can then proceed as you did originally, training your network on the new training dataset and then evaluating it and analyzing videos.

> Would using higher quality video files help here?

By downsample, do you mean reduce the resolution? In this case, generally yes, reducing the video resolution will speed up computation but reduce performance. If you have trouble spotting the mice in the lower-resolution images, it might be a good idea to train on full-size images.

**Comment by beepboop174** [comment]
@n-poulsen Hey, thank you very much for the quick reply! I just updated napari via 'pip upgrade' and still cannot add missing bodyparts when refining extracted frames. Any idea what this could be?

Edit: Just noticed that when I open the label GUI under extract outlier frames and look at help>napari info, it says  0.4.18.

**Comment by n-poulsen** [comment]
Could you run `napari --info` in your `conda` prompt and let me know the output? And did you run the full command: `pip install --upgrade napari-deeplabcut` in your `conda` prompt?

**Comment by beepboop174** [comment]
Hey, yes I ran the full command, also I updated napari itself, didnt change anything either though.

```
napari: 0.4.19.post1
Platform: Windows-10-10.0.22621-SP0
Python: 3.9.19 | packaged by conda-forge | (main, Mar 20 2024, 12:38:46) [MSC v.1929 64 bit (AMD64)]
Qt: 6.3.1
PySide6:
NumPy: 1.26.4
SciPy: 1.12.0
Dask: 2024.3.1
VisPy: 0.14.2
magicgui: 0.8.2
superqt: 0.6.2
in-n-out: 0.2.0
app-model: 0.2.5
npe2: 0.7.5

OpenGL:
  - GL version:  4.6.0 - Build 31.0.101.2114
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 3440x1440, scale 1.0

Settings path:
  - C:\Users\taylorma\AppData\Local\napari\DEEPLABCUT_4455ca27950b1ff87ed31138001f96f06d8bf918\settings.yaml
Plugins:
  - napari: 0.4.19.post1 (77 contributions)
  - napari-console: 0.0.9 (0 contributions)
  - napari-deeplabcut: 0.2.1.6 (16 contributions)
  - napari-svg: 0.1.10 (2 contributions)
```

**Comment by n-poulsen** [comment]
Thanks @beepboop174 I'm looking into this!
